File: Logger/Data/Log.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Logger.Data
{
	public class Log
	{
		public int Id { get; set; }
		public DateTime TimeStamp { get { return DateTime.Now; } }
		public DateTime? StartTime { get; set; }
		public int? Duration
		{
			get
			{
				if (!StartTime.HasValue)
				{
					return null;
				}
				return Convert.ToInt32((DateTime.Now - StartTime.Value).TotalMilliseconds);
			}
		}
		public int IdLevel { get; set; }
		public string IpAddress { get; set; }
		public string Username { get; set; }
		public string Name { get; set; }
		public string CodiceFiscaleEnte { get; set; }
		public int? IdUser { get; set; }
		public int? IdEventType { get; set; }
		public int? EntityId { get; set; }

		[MaxLength(128)]
		public string EntityName { get; set; }

		public string ApplicationIpAddress { get; set; }
		public string ApplicationHost { get; set; }
		public string ApplicationName { get; set; }
		public string SessionId { get; set; }

		[MaxLength(128)]
		public string Controller { get; set; }
		[MaxLength(128)]
		public string Action { get; set; }

		public string Method { get; set; }

		/// <summary>
		/// Indica il messagio associato al log
		/// </summary>
		public string Message { get; set; }
		/// <summary>
		/// Contiene il testo dell'eventuale eccezione
		/// </summary>
		public Exception Exception { get; set; }
		public string Parameter { get; set; }
	}
}

File: HeliosEsercizio/Logger/Data/Log.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Text

Namespace Data
    Public Class Log
        Public Property Id As Integer

        Public ReadOnly Property TimeStamp As DateTime
            Get
                Return DateTime.Now
            End Get
        End Property

        Public Property StartTime As DateTime?

        Public ReadOnly Property Duration As Integer?
            Get

                If Not StartTime.HasValue Then
                    Return Nothing
                End If

                Return Convert.ToInt32((DateTime.Now - StartTime.Value).TotalMilliseconds)
            End Get
        End Property

        Public Property IdLevel As Integer
        Public Property IpAddress As String
        Public Property Username As String
        Public Property Name As String
        Public Property CodiceFiscaleEnte As String
        Public Property IdUser As Integer?
        Public Property IdEventType As Integer?
        Public Property EntityId As Integer?
        Public Property EntityName As String
        Public Property ApplicationIpAddress As String
        Public Property ApplicationHost As String
        Public Property ApplicationName As String
        Public Property SessionId As String
        Public Property Controller As String
        Public Property Action As String
        Public Property Method As String
        Public Property Message As String
        Public Property Exception As Exception
        Public Property Parameter As String
    End Class
End Namespace
